### ISSUE TITLE
Indicators in capitalization rule do not work with title case

### ISSUE THREAD

**Issue by DominikaK** [vale-cli/vale#898]
### Check for existing issues

- [X] Completed

### Environment

OS: macOS
Install method:  Homebrew
Vale version: 3.7.1

### Describe the bug / provide steps to reproduce it

When using a `capitalization` rule, `indicators` functions correctly with a `$title` match, but not with a `$sentence` match.

The following rule (Vale Studio: https://studio.vale.sh/s/bb9890c4782f80c8c30a88007219574c):
```
extends: capitalization
message: "'%s' should be in sentence case"
level: warning
scope: heading
match: $sentence
indicators:
    - ":"
```

correctly shows `# Heading One` to be incorrect, but `# Heading: One` to be correct - capital letter is ignored after the colon indicator.

However, when I change match to `$title` (Vale Studio: https://studio.vale.sh/s/28ca0c60b15db662d53caa8375e6f47e):

```
extends: capitalization
message: "'%s' should be in title case"
level: warning
scope: heading
match: $title
indicators:
    - ":"
```

both `# Heading one` and `# Heading: one` are marked as mistakes - the colon indicator does not make the rule ignore the lowercase "one".

**Comment by jdkato** [comment]
This looks correct to me.

You can't compare `$title` and `$sentence` because the implication of `indicators` is very different in each case.

```markdown
<!-- sentence case -->
# Heading: One

<!-- not sentence case -->
# Heading: Here Is Another

<!-- sentence case -->
# Heading: Here is another
```

Is your expectation that sentence case implies that *only* the first letter is capitalized? I think that's unusual when it comes to headings.

**Comment by DominikaK** [comment]
The specific case I have in mind is a bit more complicated, I simplified it for the sake of the bug report, so let me explain now in detail:

I use title case for headings. However, headings often have additional information (to be precise, alternative anchor) presented in the following format:

`### This Is My Heading {#my-anchor}`

When I use Vale to check whether all headings correctly use title case, with the format above, the lowercase words in anchor trigger a false positive.
I should like to use `indicators` to mark that anything following `{#` should be ignored by the rule.

**Comment by jdkato** [comment]
This is a use case for [`TokenIgnores`](https://vale.sh/docs/keys/tokenignores).

**Comment by DominikaK** [comment]
Thanks you, I'll try it out. Just to be sure I understand correctly: there is no way to apply `TokenIgnores` to one scope only?

**Comment by jdkato** [comment]
No, but there's also no place you'd want non-standard markup like `{#my-anchor}` to be linted, right?
